packages/flutter/lib/src/material/floating_action_button.dart

```diff
@@ -114,67 +115,61 @@ class _DefaultHeroTag {
 class FloatingActionButton extends StatelessWidget {
   /// Creates a circular floating action button.
   ///
-  /// The [elevation], [highlightElevation], [mini], [shape], and [clipBehavior]
-  /// arguments must not be null. Additionally, [elevation],
-  /// [highlightElevation], and [disabledElevation] (if specified) must be
-  /// non-negative.
+  /// The [mini] and [clipBehavior] arguments must be non-null. Additionally,
+  /// [elevation], [highlightElevation], and [disabledElevation] (if specified)
+  /// must be non-negative.
   const FloatingActionButton({
     Key key,
     this.child,
     this.tooltip,
     this.foregroundColor,
     this.backgroundColor,
     this.heroTag = const _DefaultHeroTag(),
-    this.elevation = 6.0,
-    this.highlightElevation = 12.0,
-    double disabledElevation,
+    this.elevation,
+    this.highlightElevation,
+    this.disabledElevation,
     @required this.onPressed,
     this.mini = false,
-    this.shape = const CircleBorder(),
+    this.shape,
     this.clipBehavior = Clip.none,
     this.materialTapTargetSize,
     this.isExtended = false,
-  }) : assert(elevation != null && elevation >= 0.0),
-       assert(highlightElevation != null && highlightElevation >= 0.0),
+  }) : assert(elevation == null || elevation >= 0.0),
+       assert(highlightElevation == null || highlightElevation >= 0.0),
        assert(disabledElevation == null || disabledElevation >= 0.0),
        assert(mini != null),
-       assert(shape != null),
        assert(isExtended != null),
        _sizeConstraints = mini ? _kMiniSizeConstraints : _kSizeConstraints,
-       disabledElevation = disabledElevation ?? elevation,
        super(key: key);
 
   /// Creates a wider [StadiumBorder]-shaped floating action button with
   /// an optional [icon] and a [label].
   ///
-  /// The [label], [elevation], [highlightElevation], [clipBehavior] and
-  /// [shape] arguments must not be null. Additionally, [elevation]
-  /// [highlightElevation], and [disabledElevation] (if specified) must be
-  /// non-negative.
+  /// The [label] and [clipBehavior] arguments must non-null. Additionally,
+  /// [elevation] [highlightElevation], and [disabledElevation] (if specified)
```

Comment:
Hmm, I wonder if there should be a `,` between 2 words of `[elevation] [highlightElevation]` here?

Reply:
thanks!